imports/plugins/core/orders/client/templates/orders.js

```diff
@@ -89,22 +79,35 @@ const OrderHelper =  {
 
 Template.orders.onCreated(function () {
   this.state = new ReactiveDict();
+  this.newLimit = new ReactiveVar(2);
+  this.processingLimit = new ReactiveVar(2);
+  this.completedLimit = new ReactiveVar(2);
   this.state.setDefault({
     orders: []
   });
 
   const filterName = this.data && this.data.filter && this.data.filter.name || "new";
   Reaction.setUserPreferences(PACKAGE_NAME, ORDER_LIST_FILTERS_PREFERENCE_NAME, filterName);
 
-  // Watch for updates to the subscription and query params
-  // fetch available orders
   this.autorun(() => {
-    this.subscribe("Orders");
-    const filter = Reaction.getUserPreferences(PACKAGE_NAME, ORDER_LIST_FILTERS_PREFERENCE_NAME, DEFAULT_FILTER_NAME);
-    const query = OrderHelper.makeQuery(filter);
-    const orders = Orders.find(query).fetch();
-
-    this.state.set("orders", orders);
+    const limit = this.newLimit.get() || this.processingLimit.get() || this.completedLimit.get();
+    const subscription = this.subscribe("Orders.paginated", (limit + 1));
```

Comment:
@andela-akolawole you still have to pass a query to `Orders.find()` otherwise, it will just return every order in the DB, regardless of the query passed to the subscription.